src/main/java/gift/model/Product.java

```diff
@@ -1,9 +1,19 @@
-package gift;
+package gift.model;
+
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
 
 public class Product {
     private long id;
+
+    @NotBlank(message = "상품의 이름은 필수 항목입니다.")
+    @Size(max = 15, message = "상품의 이름은 최대 15자까지 입력할 수 있습니다.")
     private String name;
```

Comment:
```suggestion
    @NotBlank(message = "상품의 이름은 필수 항목입니다.")
    @Size(max = 15, message = "상품의 이름은 최대 15자까지 입력할 수 있습니다.")
    @Pattern(regexp = "^[\\w\\s\\(\\)\\[\\]\\+\\-\\&\\/\\_가-힣]*$", message = "상품의 이름에 사용 불가한 문자가 포함되어 있습니다.")
    private String name;
```
이런식으로도 가능합니다.
지금처러 밸리데이터를 사용하는 방식도 좋습니다~

그리고 카카오를 포함 여부도 커스텀 어노테이션을 사용해 밸리데이터 없이 어노테아션으로 처리할 수 있습니다. 추천하진 않습니다.